### PR TITLE
Hide built-in plugins from version command

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

We support built-in plugins by using the `VAST_REGISTER_PLUGIN` macro from within a source file that is part of either the `vast` or the `libvast` targets.  These plugins are always loaded and cannot be disabled. This channge hides such plugins from the version output, as they are really not plugins but simply use the plugin framework to add functionality.

The idea for this came up when reviewing the transforms functionality, which has some transformation as such built-in plugins directly as part of `libvast`.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

### CODE REVIEW

**Comment by mavam** [comment]
With the plugin information hidden, how would you figure out whether your binary supports a given feature/plugin or not?

**Comment by dominiklohmann** [comment]
> With the plugin information hidden, how would you figure out whether your binary supports a given feature/plugin or not?

To clarify, "built-in plugins" are not really plugins. They are just static part of the VAST binary or libvast library that are accessible via the plugin API and essentially work like statically linked bundled plugins. Since they're not optional there's no need to check for this, ever.

File: libvast/vast/plugin.hpp [hunk 1]
```diff
@@ -232,18 +232,27 @@ class writer_plugin : public virtual plugin {
 /// @relates plugin
 class plugin_ptr final {
 public:
+  /// The display policy of a plugin.
+  enum class visible {
+    yes, ///< Show the plugin in the output of the version command.
+    no,  ///< Hide the plugin in the output of the version command.
+  };
```
Comment (lava) [inline libvast/vast/plugin.hpp:239]: It feels a bit weird to have the plugin dictate how `vast version` behaves. What do you think about an enum like this

```
enum class plugin_type {
  dynamic,
  static,
  native
}
```

and then letting `vast version` decide itself how it renders each type of plugin?

File: libvast/vast/plugin.hpp [hunk 2]
```diff
@@ -304,6 +317,14 @@ class plugin_ptr final {
 
 #if defined(VAST_ENABLE_STATIC_PLUGINS_INTERNAL)
 
+#  if VAST_ENABLE_STATIC_PLUGINS_INTERNAL == 1
```
Comment (lava) [inline libvast/vast/plugin.hpp:320]: It feels like there should be a cleaner solution to this than having a magic number.

How about adding an additional defines like `VAST_OPTIONAL_PLUGIN`/`VAST_NATIVE_PLUGIN` that are set in the respective build contexts?

Or alternatively we at least could turn the numbers into named constants like
```
#define PLUGIN_VISIBILITY_DYNAMIC 1
```
etc.

**Review by lava — COMMENTED** [review]
While this approach would work, it feels a little bit too ad-hoc, especially since this not just some header in `detail/` but part of our public plugin API. Let's see if we can find a cleaner solution for this.

File: libvast/vast/plugin.hpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/vast/plugin.hpp:240]: I made this change, please check back.

File: libvast/vast/plugin.hpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/vast/plugin.hpp]: The logic is a bit more readable now and does not rely on magic constants any more.

**Review by lava — APPROVED** [review]
Looks good now. There's still an argument to be made that `ENABLE` in `VAST_ENABLE_STATIC_PLUGINS` is the wrong choice of words (the current semantics are more like `VAST_CURRENT_PLUGIN_TYPE`), but that's just an implementation detail in our CMakeLists so it doesn't need to hold up this PR; we can still change it anytime later if we find the motivation.

**Pull request merged**